test/agent_test.go

```diff
@@ -33,194 +32,205 @@ var (
-	defer finish()
-
-	ac1, err := newAgentClient(agentToken(), defaultServerPort)
-	require.NoError(t, err)
-	ac2, err := newAgentClient(agentToken(), defaultServerPort)
-	require.NoError(t, err)
-	ac3, err := newAgentClient(agentToken(), defaultServerPort)
-	require.NoError(t, err)
-	ac4, err := newAgentClient(agentToken(), defaultServerPort)
-	require.NoError(t, err)
-	ac5, err := newAgentClient(agentToken(), defaultServerPort)
-	require.NoError(t, err)
-	ac6, err := newAgentClient(agentToken(), defaultServerPort)
-	require.NoError(t, err)
-	defer ac1.close()
-	defer ac2.close()
-	defer ac3.close()
-	defer ac4.close()
-	defer ac5.close()
-	defer ac6.close()
-	ac1.Run(livekit.JobType_JT_ROOM, "default")
-	ac2.Run(livekit.JobType_JT_ROOM, "default")
-	ac3.Run(livekit.JobType_JT_PUBLISHER, "default")
-	ac4.Run(livekit.JobType_JT_PUBLISHER, "default")
-	ac5.Run(livekit.JobType_JT_PARTICIPANT, "default")
-	ac6.Run(livekit.JobType_JT_PARTICIPANT, "default")
-
-	testutils.WithTimeout(t, func() string {
-		if ac1.registered.Load() != 1 || ac2.registered.Load() != 1 || ac3.registered.Load() != 1 || ac4.registered.Load() != 1 || ac5.registered.Load() != 1 || ac6.registered.Load() != 1 {
-			return "worker not registered"
-		}
-
-		return ""
-	}, RegisterTimeout)
-
-	c1 := createRTCClient("c1", defaultServerPort, nil)
-	c2 := createRTCClient("c2", defaultServerPort, &testclient.Options{UseJoinRequestQueryParam: true})
-	waitUntilConnected(t, c1, c2)
-
-	// publish 2 tracks
-	t1, err := c1.AddStaticTrack("audio/opus", "audio", "micro")
-	require.NoError(t, err)
-	defer t1.Stop()
-	t2, err := c1.AddStaticTrack("video/vp8", "video", "webcam")
-	require.NoError(t, err)
-	defer t2.Stop()
-
-	testutils.WithTimeout(t, func() string {
-		if ac1.roomJobs.Load()+ac2.roomJobs.Load() != 1 {
-			return "room job not assigned"
-		}
-
-		if ac3.publisherJobs.Load()+ac4.publisherJobs.Load() != 1 {
-			return fmt.Sprintf("publisher jobs not assigned, ac3: %d, ac4: %d", ac3.publisherJobs.Load(), ac4.publisherJobs.Load())
-		}
-
-		if ac5.participantJobs.Load()+ac6.participantJobs.Load() != 2 {
-			return fmt.Sprintf("participant jobs not assigned, ac5: %d, ac6: %d", ac5.participantJobs.Load(), ac6.participantJobs.Load())
-		}
-
-		return ""
-	}, 6*time.Second)
-
-	// publish 2 tracks
-	t3, err := c2.AddStaticTrack("audio/opus", "audio", "micro")
-	require.NoError(t, err)
-	defer t3.Stop()
-	t4, err := c2.AddStaticTrack("video/vp8", "video", "webcam")
-	require.NoError(t, err)
-	defer t4.Stop()
-
-	testutils.WithTimeout(t, func() string {
-		if ac1.roomJobs.Load()+ac2.roomJobs.Load() != 1 {
-			return "room job must be assigned 1 time"
-		}
-
-		if ac3.publisherJobs.Load()+ac4.publisherJobs.Load() != 2 {
-			return "2 publisher jobs must assigned"
-		}
-
-		if ac5.participantJobs.Load()+ac6.participantJobs.Load() != 2 {
-			return "2 participant jobs must assigned"
-		}
-
-		return ""
-	}, AssignJobTimeout)
+	for _, useSinglePeerConnection := range []bool{false, true} {
```

Comment:
Using this for all tests to test both dual peer connection and single peer connection modes. Using a separate argument to `createRTCClient` helper to indicate use of single peer connection. But, it is possible to roll it into `testclient.Options`. But, wanted to use an extra argument to ensure I have covered all the places which uses this helper to test both modes. Once it is working, will probably change it to use an option.